server/__tests__/signupAPI.test.js

```diff
@@ -0,0 +1,96 @@
+import { expect } from 'chai';
+import supertest from 'supertest';
+import mongoose from 'mongoose';
+import { Mockgoose } from 'mockgoose';
+import app from '../src/app/app';
+import getMockUser from '../src/utils/testUtils';
+import { TEST_PORT, DB_URL } from '../src/app/appConstants';
+
+let mockUser;
+let testServer;
+let request;
+let mockgoose;
+
+beforeAll(() => {
+  mockgoose = new Mockgoose(mongoose);
+  mockgoose.prepareStorage().then(() => {
+    mongoose.connect(DB_URL, { useNewUrlParser: true, useCreateIndex: true });
+  });
+});
+
+beforeEach((done) => {
+  mockUser = getMockUser();
+  testServer = app.listen(TEST_PORT, () => {
+    request = supertest(testServer);
+    return done();
```

Comment:
Should we use return here? because it is an anonymous function. we are not assigning the result to anything. @nimish-gupta could you give your thoughts here?

Reply:
you can just call done() or return done(), it doesn't make much sense over here. We just need to call done which is imp.

Reply:
It seems better to me if functions have a return statement in them. I can change it to just `done()`. Do you want me to?

Reply:
I'll just remove the return there.